src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/DebtorsServiceImpl.java

```diff
@@ -0,0 +1,129 @@
+import org.springframework.stereotype.Service;
+import org.springframework.web.util.UriTemplate;
+import uk.gov.companieshouse.api.ApiClient;
+import uk.gov.companieshouse.api.error.ApiErrorResponseException;
+import uk.gov.companieshouse.api.handler.exception.URIValidationException;
+import uk.gov.companieshouse.api.model.accounts.smallfull.Debtors.DebtorsApi;
+import uk.gov.companieshouse.api.model.accounts.smallfull.SmallFullApi;
+import uk.gov.companieshouse.api.model.accounts.smallfull.SmallFullLinks;
+import uk.gov.companieshouse.web.accounts.api.ApiClientService;
+import uk.gov.companieshouse.web.accounts.exception.ServiceException;
+import uk.gov.companieshouse.web.accounts.model.smallfull.notes.debtors.Debtors;
+import uk.gov.companieshouse.web.accounts.service.company.CompanyService;
+import uk.gov.companieshouse.web.accounts.service.smallfull.DebtorsService;
+import uk.gov.companieshouse.web.accounts.transformer.smallfull.DebtorsTransformer;
+import uk.gov.companieshouse.web.accounts.util.ValidationContext;
+import uk.gov.companieshouse.web.accounts.validation.ValidationError;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class DebtorsServiceImpl implements DebtorsService {
+
+    @Autowired
+    private ApiClientService apiClientService;
+
+    @Autowired
+    private ValidationContext validationContext;
+
+    @Autowired
+    private DebtorsTransformer transformer;
+
+    @Autowired
+    private CompanyService companyService;
+
+    private static final UriTemplate SMALL_FULL_URI =
+        new UriTemplate("/transactions/{transactionId}/company-accounts/{companyAccountsId}/small-full");
+
+    private static final UriTemplate DEBTORS_URI = new UriTemplate(SMALL_FULL_URI.toString() + "/notes/debtors");
+
+    private static final String INVALID_URI_MESSAGE = "Invalid URI for debtors resource";
+
+    @Override
+    public Debtors getDebtors(String transactionId, String companyAccountsId, String companyNumber) throws ServiceException {
+        DebtorsApi debtorsApi = getDebtorsApi(transactionId, companyAccountsId);
+
+        return transformer.getDebtors(debtorsApi);
+    }
+
+    @Override
+    public List<ValidationError> submitDebtors(
+        String transactionId, String companyAccountsId,
+        Debtors debtors, String companyNumber) throws ServiceException {
+
+        ApiClient apiClient = apiClientService.getApiClient();
+
+        String uri = DEBTORS_URI.expand(transactionId, companyAccountsId).toString();
+
+        DebtorsApi debtorsApi = getDebtorsApi(transactionId, companyAccountsId);
+
+        String smallFullUri = SMALL_FULL_URI.expand(transactionId, companyAccountsId).toString();
+        SmallFullApi smallFullApi = getSmallFullData(apiClient, smallFullUri);
+        if (debtorsApi == null) {
+            debtorsApi = new DebtorsApi();
+        }
+
+        transformer.setDebtors(debtors, debtorsApi);
+
+        boolean isCreated = hasDebtors(smallFullApi.getLinks());
```

Comment:
Could rename this to be more explicit, something like `debtorsResourceExists`? That's personal preference though, it's clear enough to understand as is.